### ISSUE TITLE
Implement BatchNorm as a lowering pass

### ISSUE THREAD

**Issue by narendasan** [NVIDIA/TRTorch#54]
Right now there are a couple ways that batch norm is handled, there will be a folding pass, that merges applicable `Conv2D -> BatchNorm` patterns, there is a convertor that acts as a fallback. It would be nice if we could move that converter to be a lowering pass that synthesizes a subgraph of lower level op converters instead of having a large batch norm specific one. Some of the work is done in the unused unpack_batch_norm pass but it needs to be firmed up to support multiple dimensions.
